_capability/6-quality/index.md

```diff
@@ -20,27 +19,50 @@ Testing ensures that:
+- Tests are regularly run and often triggered by a development activity, such as modifying a source file or getting the latest code from the repository.
+
 
- - [Getting to Zero Exceptions](http://yellerapp.com/posts/2015-06-01-getting-to-exception-zero.html)
- - [Eradicating Non-Determinism in Tests](http://martinfowler.com/articles/nonDeterminism.html)
+### Working towards level 2
 
-### Level 0
-  - Automation products are used to record / describe the tests, which can be run on demand
-  - Users report more bugs then the testing process
-  - The team recognise false-positives by name, and dismiss them with "Oh, that's just the usual exception"
-  - For no reason, tests that previously work start to fail, or tests that previously fail start to work
+- Start looking into techniques such as mocking and dependency injection, to test aspects of the system that depend on external input such as databases, time of day, and other systems.
+- Explore alternative testing techniques, such as Behaviour Driven Development, to complement your approach to automated testing. Try to keep in mind that there is no _one_ correct way to test, and that your approach should be developed based on your user needs.
 
+# Level 2
 
-### Level 1
- - The project is built following TDD, and tests are run religiously throughout the development process, including on committing to the version control repository
- - There are several hard-coded integration tests, which rely on precise configurations of databases or responses from services.
- - Techniques such as MVC /  MVVM are used to decouple the business logic behind views from the presentation of views, allowing the logic to be 'headless' tested.
- - Whenever a defect is identified, an automated test is written to verify it's existence, and when the defect is fixed, the automated test ensures it never happens again
+The team have developed the confidence to experiment and innovate to help deliver the user needs. Code is continually refactored to remove technical debt, greatly accelerating the rate of delivering new features.
 
+### Indications that you're operating at level 2
 
-### Level 2
- - Code analysis tools are used to identify code coverage, which is 100% for the majority the code base
- - Automated integration and acceptance testing tools are used to ensure that the overarching system works as expected
- -  Dependency injection and mocking frameworks are used to isolate the components under-test from the rest of the system, allowing the components to be tested for a variety of circumstances independently of the dependencies, such as a database.
- - The team have a 'zero exceptions in production' mantra.
+- Code analysis tools are used to identify code coverage, which is near 100% for the majority the code base, and particularly hairy sections of the code have several tests to ensure the edge cases work.
```

Comment:
100% code coverage is often a bogus metric, I've seen codebases with 100% coverage but no actual assertions